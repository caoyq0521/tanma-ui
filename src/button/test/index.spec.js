import { mount } from '@vue/test-utils';
import TmButton from '../button.vue';

describe('tmButton', () => {
  it('render tmButton', () => {
    const wrapper = mount(TmButton);
    expect(wrapper).toMatchSnapshot();
 
  });

  it('create button', () => {
    const wrapper = mount(TmButton);

    expect(wrapper.find('.tm-button').exists()).toBe(true);
  });

  it('whether type of button', async () => {
    const wrapper = mount(TmButton);

    await wrapper.setProps({ type: 'default' });
    expect(wrapper.find('.tm-button--default').exists()).toBe(true);

    await wrapper.setProps({ type: 'primary' });
    expect(wrapper.find('.tm-button--primary').exists()).toBe(true);

    await wrapper.setProps({ type: 'danger' });
    expect(wrapper.find('.tm-button--danger').exists()).toBe(true);

    await wrapper.setProps({ type: 'addition' });
    expect(wrapper.find('.tm-button--addition').exists()).toBe(true);

    await wrapper.setProps({ type: 'forbidden' });
    expect(wrapper.find('.tm-button--forbidden').exists()).toBe(true);

    await wrapper.setProps({ type: 'wicked' });
    expect(wrapper.find('.tm-button--wicked').exists()).toBe(true);

    await wrapper.setProps({ type: 'text' });
    expect(wrapper.find('.tm-button--text').exists()).toBe(true);
  });

  it('whether disabled of button', () => {
    const wrapper = mount(TmButton, {
      propsData: {
        disabled: true
      }
    });

    expect(wrapper.find('.disabled').exists()).toBe(true);
  });

  it('whether loading of button', () => {
    const wrapper = mount(TmButton, {
      propsData: {
        loading: true
      }
    });

    expect(wrapper.find('.tm-icon-load').exists()).toBe(true);
  });
})

