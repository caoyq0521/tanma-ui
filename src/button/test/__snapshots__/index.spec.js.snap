// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tmButton render tmButton 1`] = `
<button type="default" class="tm-button tm-button--medium tm-button--default">
  <!---->
</button>
`;
